### ISSUE TITLE
Server rates are parsed wrong

### ISSUE THREAD

**Issue by farrainbow** [OpenKore/openkore#333]
```
=========================== Server Infos ===========================
EXP Rates: 200% (Base 100% + Premium 0% + Server 100000% + Plus %) 
Drop Rates: 200% (Base 100% + Premium 0% + Server 100000% + Plus %) 
Death Penalty: 100% (Base 100% + Premium 0% + Server 0% + Plus %) 
=====================================================================
```

Total and base rate is probably right, server rate is obviously wrong. That's from packet `097B`. Are there servers which actually use `08CB` for that? Many servers include both in current `recvpackets`.

**Comment by farrainbow** [comment]
Premium rate needs to be tested for a case when it's not `0`.

**Comment by farrainbow** [comment]
After fixing that, `rates_info` and `rates_info2` handlers can be merged with the help of custom packet parsers.

**Comment by s936511** [comment]
# 08CB
sub rates_info {
	my ($self, $args) = @_;
	my %rates = (
		exp => { total => $args->{exp} },
		death => { total => $args->{death} },
		drop => { total => $args->{drop} },
	);

	# get details
	for (my $offset = 0; $offset < length($args->{detail}); $offset += 7) {
		my ($type, $exp, $death, $drop) = unpack("C s3", substr($args->{detail}, $offset, 7));
		$rates{exp}{$type} = $exp; $rates{death}{$type} = $death; $rates{drop}{$type} = $drop;
	}
sub rates_info2 {
	my ($self, $args) = @_;
	my %rates = (
		exp => { total => $args->{exp}/(100*10) }, # Value to Percentage => /100
		death => { total => $args->{death}/(100*10) }, # 1 d.p. => /10
		drop => { total => $args->{drop}/(100*10) },
	);

	# get details
	for (my $offset = 0; $offset < length($args->{detail}); $offset += 13) {
		my ($type, $exp, $death, $drop) = unpack("C V3", substr($args->{detail}, $offset, 13));
		$rates{exp}{$type} = $exp/(100*10); $rates{death}{$type} = $death/(100*10); $rates{drop}{$type} = $drop/(100*10);
	}

	# we have 4 kinds of detail:
	# $rates{exp or drop or death}{DETAIL_KIND}
	# 0 = base server exp (?)
	# 1 = premium acc additional exp
	# 2 = server additional exp
	# 3 = not sure, maybe it's for "extra exp" events? never seen this using the official client (bRO)
	message T("=========================== Server Infos ===========================\n"), "info";
	message TF("EXP Rates: %s\% (Base %s\% + Premium %s\% + Server %s\% + Plus %s\%) \n", $rates{exp}{total}, $rates{exp}{0}+100, $rates{exp}{1}, $rates{exp}{2}, $rates{exp}{3}), "info";
	message TF("Drop Rates: %s\% (Base %s\% + Premium %s\% + Server %s\% + Plus %s\%) \n", $rates{drop}{total}, $rates{drop}{0}+100, $rates{drop}{1}, $rates{drop}{2}, $rates{drop}{3}), "info";
	message TF("Death Penalty: %s\% (Base %s\% + Premium %s\% + Server %s\% + Plus %s\%) \n", $rates{death}{total}, $rates{death}{0}+100, $rates{death}{1}, $rates{death}{2}, $rates{death}{3}), "info";
	message "=====================================================================\n", "info";
}

**Comment by lututui** [comment]
Fixed in #748
